joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/AstPackagePass.scala

```diff
@@ -1,28 +1,56 @@
 package io.joern.rubysrc2cpg.passes
 
 import better.files.File
-import io.joern.rubysrc2cpg.astcreation.AstCreator
 import io.joern.rubysrc2cpg.utils.{PackageContext, PackageTable}
 import io.joern.x2cpg.datastructures.Global
 import io.shiftleft.codepropertygraph.Cpg
 import io.shiftleft.passes.ConcurrentWriterCpgPass
+import org.jruby.Ruby
+import org.jruby.ast.{Colon2Node, DefnNode, Node, NodeType, ReturnNode}
 
-import scala.util.Try
+import scala.collection.mutable.ListBuffer
 
 class AstPackagePass(cpg: Cpg, tempExtDir: String, global: Global, packageTable: PackageTable, inputPath: String)
     extends ConcurrentWriterCpgPass[String](cpg) {
+
+  private val rubyInstance = Ruby.getGlobalRuntime()
   override def generateParts(): Array[String] =
     getRubyDependenciesFile(inputPath) ++ getRubyDependenciesFile(tempExtDir)
 
   override def runOnPart(diffGraph: DiffGraphBuilder, filePath: String): Unit = {
-    Try(
-      new AstCreator(
-        filePath,
-        global,
-        PackageContext(resolveModuleNameFromPath(filePath), packageTable),
-        Option(inputPath)
-      ).createAst()
-    )
+    val moduleName = resolveModuleNameFromPath(filePath)
+    try {
+      processRubyDependencyFile(filePath, moduleName)
+    } catch {
+      case ex: Exception =>
+        println(s"Error while parsing $moduleName module File ($filePath): ${ex.getMessage}")
+    }
+  }
+
+  private def fetchMethodInfoFromNode(node: Node, currentNameSpace: ListBuffer[String], moduleName: String): Unit = {
+    if (node != null) {
+      node.getNodeType match {
+        case NodeType.CLASSNODE | NodeType.MODULENODE =>
+          val classOrModuleName = node.childNodes().get(0).asInstanceOf[Colon2Node].getName.toString
```

Comment:
get(0) doesn't seem safe, handle this code

Reply:
done.